lib/cipherstash/client.rb

```diff
@@ -153,6 +155,10 @@ def create_collection(name, schema)
         }
 
         indexes = schema.fetch("indexes", {}).map do |idx_name, idx_settings|
+          if Index.invalid_unique_constraint?(idx_settings["unique"], idx_settings["kind"])
```

Comment:
I think I'd move this check into `Index.settings`.  https://github.com/cipherstash/ruby-client/blob/b6f82afb2e5c016df78e1e9b96f7a72404a72597/lib/cipherstash/index.rb#L43-L50

Rather than having the `Index.invalid_unique_constraint?` have a list of which indexes support a unique constraint, we could move it into a method on each subclass. This is similar to how we check if queries on certain indexes support ordering.


You would end up with something like:
```ruby
      def settings(name, base_settings, schema)
        subclass = subclass_from_kind(base_settings["kind"])

        if base_settings.key?("unique") && !subclass.uniqueness_supported?
          # raise
        end

        {
          meta: subclass.meta(name),
          mapping: subclass.mapping(base_settings, schema),
        }
      end
```

`Index` could also provide a default implementation of `uniqueness_supported?` that returns false so that we only need to implement it in subclasses that actually support uniqueness.

Reply:
nice, I like that setup. I'll give that a go.

Reply:
Much better, thanks for the suggestion.